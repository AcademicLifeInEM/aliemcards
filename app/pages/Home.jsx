import React, { PureComponent } from 'react';
import { post } from 'axios';

import CardList from '../cards/CardList';

export default class Home extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      newest: [],
      updated: [],
    };
  }

  componentDidMount() {
    post('/graphql', {
      query: `
        query {
          recentlyAdded {
            ...cardFragment
          }
          recentlyUpdated {
            ...cardFragment
          }
        }

        fragment cardFragment on Card {
          id
          title
          categories
        }
      `,
    })
    .then(res => {
      if (res.status !== 200) throw res.status;
      const { recentlyAdded, recentlyUpdated } = res.data.data;
      this.setState({ newest: recentlyAdded, updated: recentlyUpdated });
    })
    .catch(err => console.error(`Error: API response status code = ${err}`));
  }

  render() {
    return (
      <div className="row">
        <div className="one-half column">
          <h1>New Cards</h1>
          <CardList cards={this.state.newest} />
        </div>
        <div className="one-half column">
          <h1>Updated Cards</h1>
          <CardList cards={this.state.updated.filter(card => card.updates !== null)} />
        </div>
      </div>
    );
  }
}
