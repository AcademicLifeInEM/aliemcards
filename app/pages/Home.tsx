import * as React from 'react';
import { gql, graphql } from 'react-apollo';
import { Card } from '../../server/models/';
import CardList from '../cards/CardList';

interface Props {
  data: {
    recentlyAdded: Array<Pick<Card, 'id'|'title'|'categories'|'updates'>>;
    recentlyUpdated: Array<Pick<Card, 'id'|'title'|'categories'|'updates'>>;
    networkStatus: number;
  };
}

export const homepageCards = gql`
  query homepageCards {
    recentlyAdded: cards(orderBy: created, limit: 5) {
      ...cardFragment
    }
    recentlyUpdated: cards(orderBy: updated, limit: 5) {
      ...cardFragment
    }
  }

  fragment cardFragment on Card {
    id
    title
    categories {
      id
      name
    }
    updates
  }
`;

@graphql(homepageCards)
export default class Home extends React.PureComponent<Props, {}> {
  render() {
    const newest = this.props.data.recentlyAdded || [];
    const updated = this.props.data.recentlyUpdated || [];
    return (
      <div className="row row--wrap">
        <div className="column column--50">
          <h1>New Cards</h1>
          { newest.length > 0 &&
            <CardList cards={newest} />
          }
        </div>
        <div className="column column--50">
          <h1>Updated Cards</h1>
          { updated.length > 0 &&
            <CardList cards={updated} />
          }
        </div>
      </div>
    );
  }
}
