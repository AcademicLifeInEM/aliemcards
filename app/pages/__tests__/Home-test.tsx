jest.mock('../../cards/CardList');
jest.mock('react-apollo');

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Home from '../Home';

const data: any = {
  recentlyAdded: [],
  recentlyUpdated: [],
};

describe('<Home />', () => {
  it('should work with 0 length cards', () => {
    const component = renderer.create(<Home data={data} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should work', () => {
    const d: any = {};
    const component = renderer.create(<Home data={d} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
