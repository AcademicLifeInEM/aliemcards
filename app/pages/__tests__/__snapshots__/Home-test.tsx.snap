// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should work 1`] = `
<div
  className="row row--wrap"
>
  <div
    className="column"
  >
    <h1>
      New Cards
    </h1>
  </div>
  <div
    className="column"
  >
    <h1>
      Updated Cards
    </h1>
  </div>
</div>
`;

exports[`<Home /> should work with 0 length cards 1`] = `
<div
  className="row row--wrap"
>
  <div
    className="column"
  >
    <h1>
      New Cards
    </h1>
  </div>
  <div
    className="column"
  >
    <h1>
      Updated Cards
    </h1>
  </div>
</div>
`;
